server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatch.java

```diff
@@ -0,0 +1,458 @@
+    TransportNodesListGatewayStartedShardsBatch.Request,
+    TransportNodesListGatewayStartedShardsBatch.NodesGatewayStartedShardsBatch,
+    TransportNodesListGatewayStartedShardsBatch.NodeRequest,
+    TransportNodesListGatewayStartedShardsBatch.NodeGatewayStartedShardsBatch>
+    implements
+        AsyncBatchShardFetch.Lister<
+            TransportNodesListGatewayStartedShardsBatch.NodesGatewayStartedShardsBatch,
+            TransportNodesListGatewayStartedShardsBatch.NodeGatewayStartedShardsBatch> {
+
+    public static final String ACTION_NAME = "internal:gateway/local/started_shards_batch";
+    public static final ActionType<NodesGatewayStartedShardsBatch> TYPE = new ActionType<>(
+        ACTION_NAME,
+        NodesGatewayStartedShardsBatch::new
+    );
+
+    private final Settings settings;
+    private final NodeEnvironment nodeEnv;
+    private final IndicesService indicesService;
+    private final NamedXContentRegistry namedXContentRegistry;
+
+    @Inject
+    public TransportNodesListGatewayStartedShardsBatch(
+        Settings settings,
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NodeEnvironment env,
+        IndicesService indicesService,
+        NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(
+            ACTION_NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            Request::new,
+            NodeRequest::new,
+            ThreadPool.Names.FETCH_SHARD_STARTED,
+            NodeGatewayStartedShardsBatch.class
+        );
+        this.settings = settings;
+        this.nodeEnv = env;
+        this.indicesService = indicesService;
+        this.namedXContentRegistry = namedXContentRegistry;
+    }
+
+    @Override
+    public void list(
+        DiscoveryNode[] nodes,
+        Map<ShardId, String> shardIdsWithCustomDataPath,
+        ActionListener<NodesGatewayStartedShardsBatch> listener
+    ) {
+        execute(new Request(nodes, shardIdsWithCustomDataPath), listener);
+    }
+
+    @Override
+    protected NodeRequest newNodeRequest(Request request) {
+        return new NodeRequest(request);
+    }
+
+    @Override
+    protected NodeGatewayStartedShardsBatch newNodeResponse(StreamInput in) throws IOException {
+        return new NodeGatewayStartedShardsBatch(in);
+    }
+
+    @Override
+    protected NodesGatewayStartedShardsBatch newResponse(
+        Request request,
+        List<NodeGatewayStartedShardsBatch> responses,
+        List<FailedNodeException> failures
+    ) {
+        return new NodesGatewayStartedShardsBatch(clusterService.getClusterName(), responses, failures);
+    }
+
+    /**
+     * This function is similar to nodeoperation method of {@link TransportNodesListGatewayStartedShards} we loop over
+     * the shards here to fetch the shard result in bulk.
+     *
+     * @param request
```

Comment:
```
/Volumes/workplace/opensearch/OpenSearch/server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatch.java:138: warning: no description for @param
     * @param request
       ^
error: warnings found and -Werror specified
```

getting this exception while compiling just add a description after `request`

Reply:
Added description in new commit.